File: cld/src/main/java/cloudcmd/cld/commands/Listing.java
package cloudcmd.cld.commands;


import cloudcmd.cld.IndexStorageService;
import jpbetz.cli.Command;
import jpbetz.cli.CommandContext;
import jpbetz.cli.SubCommand;
import org.json.JSONArray;
import org.json.JSONObject;

@SubCommand(name = "ls", description = "Perform a directory listing of archived files.")
public class Listing implements Command
{
  @Override
  public void exec(CommandContext commandLine) throws Exception
  {
    JSONObject filter = new JSONObject();

    JSONArray selections = IndexStorageService.instance().find(filter);

    for (int i = 0; i < selections.length(); i++)
    {
      JSONObject entry = selections.getJSONObject(i).getJSONObject("data");
      System.out.println(entry.getString("path"));
    }
  }
}

File: cld/src/main/scala/cloudcmd/cld/commands/Listing.scala
package cloudcmd.cld.commands

import jpbetz.cli.{Command, CommandContext, Opt, SubCommand}
import cloudcmd.common.IndexedContentAddressableStorage
import cloudcmd.cld.AdapterUtil

@SubCommand(name = "ls", description = "Perform a directory listing of archived files.")
class Listing extends Command {

  @Opt(opt = "n", longOpt = "minTier", description = "min tier to verify to", required = false) var _minTier: Number = 0
  @Opt(opt = "m", longOpt = "maxTier", description = "max tier to verify to", required = false) var _maxTier: Number = Integer.MAX_VALUE
  @Opt(opt = "u", longOpt = "uri", description = "adapter URI", required = false) var _uri: String = null

  def exec(p1: CommandContext) {
    AdapterUtil.exec(p1, _uri, _minTier, _maxTier, doCommand)
  }

  def doCommand(adapter: IndexedContentAddressableStorage) {
    adapter.find() foreach { fmd => System.out.println(fmd.getPath) }
  }
}